barretenberg/cpp/src/barretenberg/sumcheck/sumcheck.hpp

```diff
@@ -190,13 +188,11 @@ template <typename Flavor> class SumcheckProver {
     SumcheckOutput<Flavor> prove(ProverPolynomials& full_polynomials,
                                  const bb::RelationParameters<FF>& relation_parameters,
                                  const RelationSeparator alpha,
-                                 const std::vector<FF>& gate_challenges)
+                                 const std::vector<FF>& gate_challenges,
+                                 ZKSumcheckData<Flavor> zk_sumcheck_data = ZKSumcheckData<Flavor>())
```

Comment:
decoupled the creation of a ZKSumcheckData object from the prove method. now it's taken care by a prover that uses a constructor to create these data and then feeds it to the prove method